pydantic/schema.py

```diff
@@ -502,6 +501,8 @@ def field_type_schema(
         )
         definitions.update(f_definitions)
         nested_models.update(f_nested_models)
+    else:
+        raise AssertionError('Unknown shape: %i' % field.shape)
```

Comment:
f-string please. Also `ValueError` would probably make more sense.

I also suspect this line has no coverage which is why the coverage has gone down.

Reply:
you could also probable provide a more informative error message.

Reply:
I'm honestly not sure how to trigger this assertion now generics are fixed, as I'm not sure what cases are no longer handled. I simply moved it from a check inside the `else` block to be the `else` block. Which I think aids the readability.

Reply:
pydantic has 100% coverage, I would like to keep it that way.

If we really can't reach this could, it shouldn't exist. But I would be confident we can if we really try.

Reply:
you just need to call the method directly with a custom `ModelField` which has a `shape` which pydantic doesn't recognise.

Reply:
I fiddled with this for a bit, but the test seemed rather fragile and contrived. I reverted the code back to the way it was which is easier to test.